modules/form-field/react/lib/Combobox.tsx

```diff
@@ -0,0 +1,317 @@
+    const autoCompleteItemCount = this.props.autocompleteItems.length;
+    const firstItem = 0;
+    const lastItem = autoCompleteItemCount - 1;
+    let nextIndex = null;
+
+    switch (event.key) {
+      case 'ArrowUp':
+        const upIndex = currentIndex != null ? currentIndex - 1 : lastItem;
+        nextIndex = upIndex < 0 ? lastItem : upIndex;
+        break;
+
+      case 'ArrowDown':
+        const downIndex = currentIndex != null ? currentIndex + 1 : firstItem;
+        nextIndex = downIndex >= autoCompleteItemCount ? firstItem : downIndex;
+        break;
+
+      case 'Escape':
+        this.resetSearchInput();
+        break;
+
+      case 'Enter':
+        if (this.state.selectedAutocompleteIndex != null) {
+          const item = this.props.autocompleteItems[this.state.selectedAutocompleteIndex];
+          this.handleAutocompleteClick(event, item.props);
+          event.stopPropagation();
+          event.preventDefault();
+        }
+        break;
+
+      default:
+    }
+    this.setState({ selectedAutocompleteIndex: nextIndex });
+  };
+
+  handleSearchInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    if (this.props.onChange) {
+      this.props.onChange(event)
+    }
+    this.setState({ value: event.target.value });
+  };
+
+  renderChildren = (formElement: React.ReactElement<FormFieldProps>): React.ReactNode => {
+    if (!React.isValidElement<FormFieldProps>(formElement)) {
+      return formElement
+    }
+    const newFormElementProps: Partial<FormFieldProps> = {inputId: `${this.props.accessibleId}-${labelIdPart}`}
+
+    if (this.props.grow) {
+      newFormElementProps.grow = this.props.grow;
+    }
+
+    if (React.Children.only(formElement.props.children)) {
+      newFormElementProps.children = React.Children.map(formElement.props.children, (inputElement: React.ReactElement<TextInputProps>) => {
+        if (!React.isValidElement<TextInputProps>(inputElement)) {
+          return inputElement
+        }
+        const newTextInputProps: Partial<TextInputProps> = {
+          value: this.state.value,
+          inputRef: this.inputRef,
+          'aria-autocomplete': 'list',
+          'aria-controls': `${this.props.accessibleId}-${listBoxIdPart}`,
+          'aria-activedescendant': this.state.selectedAutocompleteIndex != null ? `${this.props.accessibleId}-${optionIdPart}-${this.state.selectedAutocompleteIndex}` : '',
+          onChange: this.handleSearchInputChange,
+          onKeyDown: this.handleKeyboardShortcuts,
+          onFocus: this.setFocus,
+          onBlur: this.setBlur,
+        }
+        return React.cloneElement(inputElement, { ...inputElement.props, ...newTextInputProps });
+      })[0]
+    }
+
+    return React.cloneElement(formElement, {...formElement.props, ...newFormElementProps});
+  }
+
+  render() {
+    const {
+      accessibleId,
+      autocompleteItems,
+      children,
+      grow,
+      onChange,
```

Comment:
Missing `initialId` here. We should be pulling out all props the component accepts even if they're not used so `elemProps` is explicitly extra props passed that we're not expecting.